cmd/kube-apiserver/app/server.go

```diff
@@ -77,6 +79,14 @@ func init() {
 // NewAPIServerCommand creates a *cobra.Command object with default parameters
 func NewAPIServerCommand() *cobra.Command {
 	s := options.NewServerRunOptions()
+
+	featureGate := utilfeature.DefaultMutableFeatureGate
+	featureGate.DeferErrorsToValidation(true)
+	effectiveVersion := utilversion.DefaultEffectiveVersionRegistry.EffectiveVersionForOrRegister(
+		utilversion.ComponentGenericAPIServer, utilversion.K8sDefaultEffectiveVersion())
+	s.GenericServerRunOptions.FeatureGate = featureGate
+	s.GenericServerRunOptions.EffectiveVersion = effectiveVersion
```

Comment:
Introduce something like `options.NewServerRunOptionsWith(featureGate, effectiveVersion)` ?  (should DeferErrorsToValidation be called inside this function?)

Reply:
`DeferErrorsToValidation` is a mutable change.

Reply:
Ah, right, the defer call needs to be here.

Reply:
done